rules/repositories.bzl

```diff
@@ -149,7 +151,7 @@ def _rules_ios_bzlmod_dependencies():
             ],
             sha256 = "c6966ec828da198c5d9adbaa94c05e3a1c7f21bd012a0b29ba8ddbccb2c93b0d",
         )
-    elif bazel_version.major == "6" and bazel_version.minor == "1":
```

Comment:
@thiagohmcruz and I discussed offline - This feels like a weird special behavior to have for handling the CI use-case ([per this comment](https://github.com/bazel-ios/rules_ios/pull/773/files#r1362497302)). I like that the proposed change makes the rules_apple 2.x support a bit more explicit in the API contract. Before this change, someone using 6.1 and accidentally getting some weird default dependencies would probably be confusing.

I think we're going to omit the Bazel version check here and have this be defaults that we know work for rules_apple 2.x based on an explicit `use_rules_apple_2 = True` param. We're also going to reset this back to the values used prior to #773 (rules_apple v2.5.0 and rules_swift v1.10.0) as the defaults. For CI, Thiago is going to mutate the WORKSPACE file in the CI script to pass `use_rules_apple_2 = True`. Separately to this, we could bump everything to run on Bazel 6.4 as well and things should continue to work.

Reply:
Thank you both, that's easier to reason about